services/motion/builtin/builtin.go

```diff
@@ -282,14 +282,14 @@ func (ms *builtIn) MoveOnGlobe(
 		return false, errors.New("destination cannot be nil")
 	}
 
-	moveRequest, err := ms.newMoveOnGlobeRequest(ctx, componentName, destination, movementSensorName, obstacles, motionCfg, extra)
+	mr, err := ms.newMoveOnGlobeRequest(ctx, componentName, destination, movementSensorName, obstacles, motionCfg, extra)
 	if err != nil {
 		return false, err
 	}
 
 	// start a loop that plans every iteration and exits when something is read from the success channel
-	for {
-		ma := newMoveAttempt(ctx, moveRequest)
+	for i := 0; i < mr.maxReplans; i++ {
```

Comment:
Why do we want to limit the number of max replans?

Reply:
I doubt we want to infinite loop trying things that aren't working forever; there's probably a good number to use here.

Reply:
Since this wasn't scoped, I'm going to leave this in but update the default to `math.Inf(1)`